testing/constraints-3.7.txt

```diff
@@ -7,11 +7,16 @@
 # Then this file should have foo==1.14.0
 db-dtypes==0.3.0
 geopandas==0.9.0
-google-api-core==1.31.5
-google-cloud-bigquery-storage==2.6.0
-google-cloud-core==1.6.0
-google-resumable-media==0.6.0
+google-api-core==2.17.1
```

Comment:
Looks like we updated these without updating the corresponding minimum versions in `setup.py`. I'll try to revert as much as I can. Do we know why we needed to upgrade these?